android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java

```diff
@@ -270,16 +274,31 @@ public static Keyboard getKeyboard(JSONObject json, String packageID, int index)
   }
 
   /**
-   * Parse a kmp.json JSON object and return the keyboard count. Lexical model packages return 0
+   * Parse a kmp.json JSON object and return the touch-layout keyboard count. Lexical model packages return 0
    * @param json kmp.json as a JSON object
-   * @return int of number of keyboards. 0 if not found
+   * @return int of number of touch-layout keyboards. 0 if not found
    */
   public static int getKeyboardCount(JSONObject json) {
     int count = 0;
     try {
       if (!json.has(PP_KEYBOARDS_KEY)) {
         return count;
       }
+      // Verify JS file is in the kmp
+      boolean jsKeyboardExists = false;
+      JSONArray files = json.getJSONArray(PP_FILES_KEY);
+      for (int i = 0; i < files.length(); i++) {
+        JSONObject file = files.getJSONObject(i);
+        String filename = file.getString(PP_FILES_NAME_KEY);
+        if (FileUtils.hasJavaScriptExtension(filename)) {
+          jsKeyboardExists = true;
+          break;
+        }
+      }
+      if (!jsKeyboardExists) {
+        return count;
+      }
+
```

Comment:
Instead of just looking for a .js file (which could be, for example, part of a welcome.htm), we should check that there is an associated .js for each of the entries in the `keyboards` array.

Reply:
I changed `getKeyboardCount()` to count the number of keyboard ID's with a matching .js file.

It's possible for a keyboard ID to have a .kmx file but not .js file in the package.
During the actual package installation, only the keyboard entries where `touchKeyboardExists()` (l. 194) get processed.